sql/updates/world/3.3.5/2020_99_99_00_world.sql

```diff
@@ -0,0 +1,36 @@
+(@ENTRY, 0, 2, 0, 6, 0, 100, 0, 0, 0, 0, 0, 34, 3, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, "On death - Set instance data #3 to 3 // On just died - Set instance data #3 to 3 (DONE)");
+
+-- Gameobject Door Lever 101832 SAI and change door entry
+SET @ENTRY := 101832;
+SET @GUID := (SELECT `guid` FROM `gameobject` WHERE `id` = 17153 AND `map` = 36 AND `position_x` > -295 AND `position_x` < -285 AND `position_y` > -540 AND `position_y` < -530 AND `position_z` > 45 AND `position_z` < 55);
+SET @DOORENTRY := 16400;
+UPDATE `gameobject` SET `id` = @DOORENTRY WHERE `guid` = @GUID AND `id` = 17153;
+UPDATE `gameobject_template` SET `AIName`="SmartGameObjectAI" WHERE `entry`= @ENTRY;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=1;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(@ENTRY, 1, 0, 0, 70, 0, 100, 0, 2, 0, 0, 0, 9, 0, 0, 0, 0, 0, 0, 14, @GUID, @DOORENTRY, 0, 0, 0, 0, 0, "On loot state changed to GO_ACTIVATED - Gameobject 26185: Activate (only gameobject; sets loot state to GO_READY) // Door Lever - On Gameobject State Changed - Activate Gameobject");
+
+-- Gameobject Door Lever 101834 SAI and change door entry
+SET @ENTRY := 101834;
+SET @GUID := (SELECT `guid` FROM `gameobject` WHERE `id` = 17153 AND `map` = 36 AND `position_x` > -175 AND `position_x` < -160 AND `position_y` > -585 AND `position_y` < -575 AND `position_z` > 10 AND `position_z` < 20);
+SET @DOORENTRY := 16399;
+UPDATE `gameobject` SET `id` = @DOORENTRY WHERE `guid` = @GUID AND `id` = 17153;
+UPDATE `gameobject_template` SET `AIName`="SmartGameObjectAI" WHERE `entry`= @ENTRY;
+DELETE FROM `smart_scripts` WHERE `entryorguid`=@ENTRY AND `source_type`=1;
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(@ENTRY, 1, 0, 0, 70, 0, 100, 0, 2, 0, 0, 0, 9, 0, 0, 0, 0, 0, 0, 14, @GUID, @DOORENTRY, 0, 0, 0, 0, 0, "On loot state changed to GO_ACTIVATED - Gameobject 26182: Activate (only gameobject; sets loot state to GO_READY) // Door Lever - On Gameobject State Changed - Activate Gameobject");
```

Comment:
Just compared current objects and sniffed and they are indeed messed up, just change everything related to doors to:
```sql
UPDATE `gameobject` SET `id` = 16400 WHERE `guid` = 26185 AND `id` = 17153;
UPDATE `gameobject` SET `id` = 16399 WHERE `guid` = 26182 AND `id` = 17153;
UPDATE `smart_scripts` SET `target_type` = 20, `target_param1` = 16400, `target_param2` = 30 WHERE `entryorguid` = 101832 AND `source_type` = 1 AND `id` = 0;
UPDATE `smart_scripts` SET `target_type` = 20, `target_param1` = 16399, `target_param2` = 30 WHERE `entryorguid` = 101834 AND `source_type` = 1 AND `id` = 0;
```

Reply:
We want to avoid use of targets with guids if it is possible because guids are not stable and can be changed without script change